### PR TITLE
feat: rework documentation for vitepress

### PR DESCRIPTION
**Pull request by aldy505**
We want to streamline the documentation to be better for everyone, since we are missing too much documentation on how to do things correctly, and most of contributors don't work the same way.

This PR lays the groundwork for the documentation site. I use vitepress after considering a few other alternaitves:
- Vitepress
- Rspress (requires Rust, not everyone is okay with the bulkyness of it)
- Docusaurus (is not as lightweight as Vitepress)
- https://docs.page/

My own consideration includes:
- It's easy to deploy on GitHub Pages, or at least there are some written documentation for it
- As minimum configuration as possible, so people would be focus and spend their time more on writing markdown documentation

### CODE REVIEW

File: docs/application/architecture.md [hunk 1]
```diff
@@ -14,15 +14,16 @@ Excalidraw link: https://excalidraw.com/#json=JPBApysdZoQE1j0Tm4UBF,WbBCS-GteLqg
 - Analytics service —> Listen to Kafka/Redpanda for anything, and insert into an [OLAP database](https://www.ibm.com/topics/olap)
 - Recommendation service (optional) —> Acquire user's movement from OLAP database over time, and create a product recommendation based on the behavior
 - Promotion service —> handles coupons and automatic applied promotion on user's shopping cart
-- Image proxy service (optional) —> Convert image from blob storage to some relevant versions (black and white for sold out, etc)
+- Image proxy service (optional) —> Convert image from blob storage to some relevant versions (black and white for sold
+  out, etc.)
```
Comment (elianiva) [inline docs/application/architecture.md:18]: ini jadi newline?

File: docs/application/new-backend-service.md [hunk 2]
```diff
@@ -9,7 +9,9 @@ The team will reject and hard delete the repository if a service was made withou
 A few requirements has to be met to create a service:
 
 1. Use gRPC for presentation/transport layer. Submit your `.proto` file on the [proto repository](https://github.com/TokoBapak/proto).
-2. No third party dependency (database, message bus, blob storage, etc) that use Java (and JVM derivatives), please. Go search for an alternative. Example: Typesense instead of Elasticsearch, Redpanda instead of Kafka, and ScyllaDB instead of Cassandra.
+2. No third party dependency (database, message bus, blob storage, etc.) that use Java (and JVM derivatives), please. Go
```
Comment (elianiva) [inline docs/application/new-backend-service.md:12]: let me be a grammar nazi for a bit

```suggestion
2. No third party dependency (database, message bus, blob storage, etc.) that uses Java (and JVM derivatives), please. Go
```

**Review by elianiva — APPROVED** [review]
LGTM, cuma beberapa comment

**Review by oinpentuls — APPROVED** [review]

File: docs/application/architecture.md (hunk 1 above)
Comment (aldy505) [reply docs/application/architecture.md:18]: Dari formatternya Jetbrains

**Review by YogiPristiawan — APPROVED** [review]

**Pull request merged**